ui/src/store/resources.ts

```diff
@@ -0,0 +1,139 @@
+import { types, getEnv, flow, Instance } from 'mobx-state-tree';
+import fuzzysort from 'fuzzysort';
+import { Api } from '../api';
+import { ICategoryStore, Tag, ITag } from './category';
+import { IKindStore, Kind } from './kind';
+import { ICatalogStore, CatalogType } from './catalog';
+
+const Catalog = types.model('Catalog', {
+  id: types.number,
+  type: types.string
+});
+
+const ResourceVersion = types.model('Version', {
+  id: types.number,
+  version: types.optional(types.string, '0.1'),
+  displayName: types.string,
+  description: types.string,
+  minPipelinesVersion: types.optional(types.string, ''),
+  rawURL: types.string,
+  webURL: types.string,
+  updatedAt: types.string
+});
+
+export const Resource = types.model('Resource', {
+  id: types.number,
+  kind: types.string,
+  name: types.string,
+  catalog: Catalog,
+  latestVersion: ResourceVersion,
+  tags: types.optional(types.array(Tag), []),
+  rating: 0
+});
+
+export type IResource = Instance<typeof Resource>;
+
+export const ResourceStore = types
+  .model('ResourceStore', {
+    resources: types.array(Resource),
+    isLoading: true,
+    searchInput: types.optional(types.string, ''),
```

Comment:
can we have an instance export of `ResourceStore`
```
export type IResourceStore = Instance<typeof ResourceStore>;
```

Reply:
As we have not used it anywhere may be once the ui component needs it then we can create an interface of it